types/encoding.go

```diff
@@ -38,12 +38,14 @@ func ParseUint256orHex(val *string) (*big.Int, error) {
 		base = 16
 	}
 
-	b, ok := new(big.Int).SetString(str, base)
-
-	if !ok {
-		return nil, fmt.Errorf("could not parse")
+	floatValue, _, err := big.ParseFloat(str, base, 0, big.ToNearestEven)
```

Comment:
This change is introduced in order to be able to provide string in scientific notation, e.g. 1e24 (because it is more verbose than hexadecimal notation).